ga4gh/rnaseq2ga.py

```diff
@@ -246,78 +243,40 @@ def writeExpressionTable(writer, data):
         writer.writeExpression(rnaQuantId, quantfile)
 
 
-def rnaseq2ga(dataFolder, sqlFilename, repoPath="ga4gh-example-data/repo.db",
-              controlFile="rna_control_file.tsv", featureType="gene"):
+def rnaseq2ga(quantificationFilename, sqlFilename, localName, rnaType, dataset=None,
+              featureType="gene", description="", programs="", annotationIds="",
+              readGroupSetNames=""):
     """
     Reads RNA Quantification data in one of several formats and stores the data
     in a sqlite database for use by the GA4GH reference server.
 
-    Quantifications are specified in a tab delimited control file with columns:
-    rna_quant_name    filename        type    feature_set_name
-    read_group_set_name     description    programs
-
-    Supports the following quantification output type:
+    Supports the following quantification output types:
     Cufflinks, kallisto, RSEM
     """
-    controlFilePath = os.path.join(dataFolder, controlFile)
-    dataRepo = data_repo(repoPath)
-    rnaDB = RNASqliteStore(sqlFilename, repoPath)
-    with open(controlFilePath, "r") as rnaDatasetsFile:
-        rnaDatasetsFile.readline()  # skip header
-        for line in rnaDatasetsFile:
-            fields = line.split("\t")
-            rnaType = fields[2]
-            dataset = dataRepo.getDatasets()[0]
-            annotationId = fields[3].strip()
+    rnaDB = RNASqliteStore(sqlFilename, dataset)
+    if rnaType == "cufflinks":
+        writer = CufflinksWriter(rnaDB, featureType)
+    elif rnaType == "kallisto":
+        writer = KallistoWriter(rnaDB, featureType)
+    elif rnaType == "rsem":
+        writer = RsemWriter(rnaDB, featureType)
+    else:
+        raise Exception("Unknown RNA file type: {}".format(rnaType))
+    readGroupSetName = readGroupSetNames.strip().split(",")[0]
+    featureSetIds = ""
+    readGroupIds = ""
+    if dataset:
+        featureSetIdList = []
+        for annotationId in annotationIds.split(","):
```

Comment:
As these are feature set names we ought to remove mention of IDs and replace with featureSetNames.